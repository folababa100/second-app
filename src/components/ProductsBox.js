import React from 'react';
import axios from 'axios';
import Products from './Products';
import Header from './Header';

export default class PostsBox extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      data: []
    }
    this.loadProductsFromServer = this.loadProductsFromServer.bind(this);
  }
  loadProductsFromServer() {
    axios.get('http://localhost:3002/products', {
      headers: {
        "x-auth": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI1Yjg1NjZjOTA1MmI0MDJjODBmZjJjYWYiLCJhY2Nlc3MiOiJhdXRoIiwiaWF0IjoxNTM1NDY5MjU4fQ.9qEVqieslHkc-ZeLGW5Nv-Q3pSsat7NobFJAvW-T23g"
      }
    })
      .then(res => {
        this.setState({ data: res.data.products })
      })
  }
  componentDidMount() {
    this.loadProductsFromServer()
  }
  render() {
    return (
      <div>
        <Header/>
        <h1>Hello</h1>
        <Products data={this.state.data}/>
      </div>
    )
  }
}
