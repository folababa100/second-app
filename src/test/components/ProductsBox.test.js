import React from 'react';
import { shallow } from 'enzyme';
import ProductsBox from '../../components/ProductsBox';

test('should render component correctly', () => {
  const wrapper = shallow(<ProductsBox/>);
  expect(wrapper).toMatchSnapshot();
})
