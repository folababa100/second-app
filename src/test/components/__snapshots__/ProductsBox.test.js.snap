// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PostsBox />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Example />,
        <h1>
          Hello
</h1>,
        <Products
          data={Array []}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Hello",
        },
        "ref": null,
        "rendered": "Hello",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "data": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Example />,
          <h1>
            Hello
</h1>,
          <Products
            data={Array []}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Hello",
          },
          "ref": null,
          "rendered": "Hello",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "data": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
